matrix/main.py

```diff
@@ -132,6 +132,13 @@ def setup(matrix, args=None):
     parser.add_argument("-k", "--keep-models", action="store_true",
                         default=False, help="Keep the created per-test models "
                                             "for further inspection")
+    parser.add_argument("-r", "--deployer", default="python-libjuju",
+                        help="Method to use to deploy charms. Defaults to "
+                        "'python-libjuju'. Other options include 'congure-up'."
+                        " Note that if you choose conjure-up, --path "
+                        "should point to a conjure-up spell "
+                        "(with metadata.yaml) "
+                        "instead of to a vanilla juju bundle.")
```

Comment:
@johnsca The tricky bit about doing it automatically is that a spell is basically just a bundle with a metadata.yaml file in it. Branching just on there being (or not being) a metadata.yaml felt iffy, especially as the conjure-up docs were vague about what exactly needed to be in the file for it to be a legit spell.

I went with the switch, though technically I'd be slightly happier with automagic.